web/src/lib/components/listen-box.tsx

```diff
@@ -138,7 +138,9 @@ export default class ListenBox extends Component<Props, State> {
           <Icon type="x"/>Nope.</a>
       </div>
       <audio className="audio-box"
-        src={this.props.src}
+        // Only include the src attribute if the source is defined
+        // (empty src attributes are invalid)
+        {...this.props.src && {src: this.props.src}}
```

Comment:
I'm a bit new to (p)react; why do we need the `...` here? Is it the TS spread operator?

Reply:
In general, I agree with @nevik that this is hard to read. But given that you have a comment explaining what this does, and the footprint is very small I'm going to allow it.